src/main/java/com/salesforce/storm/spout/dynamic/metrics/DropwizardRecorder.java

```diff
@@ -0,0 +1,147 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link MetricsRecorder} implementation for Storm 1.2+ based off DropwizardMetrics.
+ *
+ * This implementation will NOT work with Storm 1.1 or older.  If you are running a pre-1.2 version please checkout {@link StormRecorder}.
+ */
+public class DropwizardRecorder implements MetricsRecorder {
+
+    private Map<String, Object> spoutConfig;
+    private TopologyContext topologyContext;
+
+    private KeyBuilder keyBuilder;
+    private TimerManager timerManager = new TimerManager();
+
+    private final Map<MetricDefinition, Timer> timers = new HashMap<>();
+    private final Map<MetricDefinition, Counter> counters = new HashMap<>();
+    private final Map<MetricDefinition, Histogram> histograms = new HashMap<>();
+
+    @Override
+    public void open(final Map<String, Object> spoutConfig, final TopologyContext topologyContext) {
+        // Store the spout config for later use
+        this.spoutConfig = spoutConfig;
+        // Store the TopologyContext for later use when we need to register metrics.
+        this.topologyContext = topologyContext;
+
+        this.keyBuilder = new KeyBuilder(getMetricRecorderTaskIdPrefix());
+
+    }
+
+    private String getMetricRecorderTaskIdPrefix() {
+        // We allow this option so that if you're using Graphite you can keep your key total down
+        if (!spoutConfig.containsKey(SpoutConfig.METRICS_RECORDER_ENABLE_TASK_ID_PREFIX)) {
+            return null;
+        }
+
+        // We have the key, assuming it's a Boolean we'll return it
+        final Object taskIdCfgValue = spoutConfig.get(SpoutConfig.METRICS_RECORDER_ENABLE_TASK_ID_PREFIX);
+        if (taskIdCfgValue instanceof Boolean && (Boolean) taskIdCfgValue) {
+            return "task" + topologyContext.getThisTaskIndex();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void countBy(MetricDefinition metric, long incrementBy, Object... metricParameters) {
+        if (!counters.containsKey(metric)) {
+            counters.put(metric, topologyContext.registerCounter(keyBuilder.build(metric, metricParameters)));
+        }
+
+        counters.get(metric).inc(incrementBy);
+    }
+
+    /**
+     * Assign an arbitrary value.
+     *
+     * This implementation uses a {@link Histogram} because we felt it was closest to the arbitrary value assignment previously done with
+     * the {@link MultiAssignableMetric}.
+     *
+     * This particularly method should be considered experimental at this point as we may revise it before finalizing the release.
+     *
+     * @param metric metric definition.
+     * @param value value to be assigned.
+     * @param metricParameters when a {@link MetricDefinition} supports interpolation on it's key, for example "foo.{}.bar" the {}
+     */
+    @Override
+    public void assignValue(MetricDefinition metric, Object value, Object... metricParameters) {
```

Comment:
So I would have thought that a assigning a value would be using a gauge under the hood?